src/rust_pub_sub/src/publisher.rs

```diff
@@ -15,19 +15,21 @@ fn main() -> Result<(), Error> {
 
     let mut publish_count: u32 = 0;
 
-    let mut cam = videoio::VideoCapture::new_default(0)?;
+
+    let mut cam = videoio::VideoCapture::new(0, videoio::CAP_ANY)?;
+    
     if !videoio::VideoCapture::is_opened(&cam)? {
-		panic!("Unable to open default camera!");
-	}
+		    panic!("Unable to open default camera!");
```

Comment:
Recommendation: avoid panic and friends.

panics are EXTREMELY difficult to handle in calling code.
You are already returning a `Result` type from this function, I recommend returning an error type here instead of panicing.

As a recommendation, [thiserror](https://crates.io/crates/thiserror) is very useful for building error types.